src/io/video.js

```diff
@@ -97,39 +80,18 @@ class Video {
      * @return {Promise.<Video>} resolves a promise to this IO device when video is ready.
      */
     enableVideo () {
-        this.enabled = true;
-        return this._setupVideo();
+        if (!this.provider) return null;
+        return this.provider.enableVideo().then(() => this._setupPreview());
     }
 
     /**
      * Disable video stream (turn video off)
+     * @return {void}
      */
     disableVideo () {
-        this.enabled = false;
-        // If we have begun a setup process, call _teardown after it completes
-        if (this._singleSetup) {
-            this._singleSetup
-                .then(this._teardown.bind(this))
-                .catch(err => this.onError(err));
-        }
-    }
-
-    /**
-     * async part of disableVideo
-     * @private
-     */
-    _teardown () {
-        // we might be asked to re-enable before _teardown is called, just ignore it.
-        if (this.enabled === false) {
-            this._disablePreview();
-            this._singleSetup = null;
-            // by clearing refs to video and track, we should lose our hold over the camera
-            this._video = null;
-            if (this._track) {
-                this._track.stop();
-            }
-            this._track = null;
-        }
+        if (!this.provider) return null;
+        this._disablePreview();
+        this.provider.disableVideo();
```

Comment:
Yeah I was trying to figure out where I wanted that call.

My thought is that disable preview does still depend on the provider, because the things that are getting disabled only get set up if there was a provider... but I guess it's possible for a developer to switch providers or set their provider back to null between their calls to enableVideo and disableVideo. Perhaps switching from an existing provider to another one or to null should result in a call to disableVideo on the previous provider before that switch is allowed to happen...?

Reply:
It doesn't super matter if we disable preview (it would just early exit once per frame if the video gets stopped but we don't know about it)

It still makes sense to disablePreview before the provider check though I think as @cwillisf said.

Reply:
> Perhaps switching from an existing provider to another one or to null should result in a call to disableVideo on the previous provider before that switch is allowed to happen...?

That sounds right to me, but I'm also starting to feel a little YAGNI... hmm.